apps/web/src/hooks/Trades.ts

```diff
@@ -148,17 +142,12 @@ export function useTradeExactOut(
           null
         )
       }
-      // search through trades with varying hops, find best trade out of them
-      let bestTradeSoFar: Trade<Currency, Currency, TradeType> | null = null
-      for (let i = 1; i <= MAX_HOPS; i++) {
-        const currentTrade =
-          Trade.bestTradeExactOut(allowedPairs, currencyIn, currencyAmountOut, { maxHops: i, maxNumResults: 1 })[0] ??
-          null
-        if (isTradeBetter(bestTradeSoFar, currentTrade, BETTER_TRADE_LESS_HOPS_THRESHOLD)) {
-          bestTradeSoFar = currentTrade
-        }
-      }
-      return bestTradeSoFar
+
+      const bestTrades =
+        Trade.bestTradeExactOut(allowedPairs, currencyIn, currencyAmountOut, { maxHops: MAX_HOPS, maxNumResults: 10 });
+      
+      const bestTrade = bestTrades.length > 0 ? compareBestTradeHops(bestTrades, MAX_HOPS) : null;
```

Comment:
@chef-jojo, @theharv3st,  sorry for the tag, I know you are really busy with the V3 maintenance work. Just want to know how soon you'll be available to take a look at this so it can be merged.